src/components/Main/Main.js

```diff
@@ -8,69 +8,67 @@ import SearchResults from '../SearchResults/SearchResults';
 import AboutMe from '../AboutMe/AboutMe';
 import PopupWithForm from '../PopupWithForm/PopupWithForm';
 import { usePopups, popupActions } from '../../contexts/PopupContext';
-import { useAuth } from '../../contexts/AuthContext';
 import AuthForm from '../AuthForm/AuthForm';
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import NothingFound from '../NothingFound/NothingFound';
+import ConnectionError from '../ConnectionError/ConnectionError';
+import { MAX_MOBILE_SIZE } from '../../utils/constants';
 
-const Main = () => {
-  const isMobileSized = useWindowSize().width < 650;
+const Main = ({ handleSignin, handleSignup, responseError, setResponseError }) => {
+  const isMobileSized = useWindowSize().width < MAX_MOBILE_SIZE;
   const [popupState, popupDispatch] = usePopups();
-  const { signIn } = useAuth();
   const [isSearching, setIsSearching] = useState(false);
   const [searchResults, setSearchResults] = useState([]);
+  const [searchKeyword, setSearchKeyword] = useState('');
   const [nothingFound, setNothingFound] = useState(false);
+  const [connectionError, setConnectionError] = useState(false);
 
   const showSignUp = () => {
+    setResponseError(null);
     popupDispatch(popupActions.closeAll);
     popupDispatch(popupActions.openSignUpPopup);
   };
 
   const showSignIn = () => {
+    setResponseError(null);
     popupDispatch(popupActions.closeAll);
     popupDispatch(popupActions.openSignInPopup);
   };
 
-  const handleSignup = () => {
-    popupDispatch(popupActions.openSuccessPopup);
-    popupDispatch(popupActions.closeSignUpPopup);
-  };
-
-  const handleSignin = () => {
-    // temporary - pre APIs
-    signIn('Elise');
-    popupDispatch(popupActions.closeSignInPopup);
-  };
-
-  const handleSearchSubmit = (results) => {
-    // temporarily mocking search results
-    setIsSearching(true);
+  const handleSearchSubmit = (results, keyword) => {
+    setSearchKeyword(keyword);
     setNothingFound(false);
     setSearchResults([]);
-    new Promise((resolve) => {
-      setTimeout(resolve, 1500);
-    }).then(() => {
-      if (!results || results.length === 0) {
-        setNothingFound(true);
-      } else {
-        setSearchResults(results);
-      }
-      setIsSearching(false);
-    });
+    if (!results || results.length === 0) {
+      setNothingFound(true);
+    } else {
+      setSearchResults(results);
+      localStorage.setItem('searchResults', JSON.stringify({ keyword, results }));
+    }
   };
 
+  useEffect(() => {
+    const searchResults = localStorage.getItem('searchResults');
+    if (searchResults) {
+      const { results, keyword } = JSON.parse(searchResults);
+      setSearchResults(results);
+      setSearchKeyword(keyword);
+    }
+  }, []);
+
   return (
     <>
       {popupState.isSigninPopupOpen && (
         <PopupWithForm
```

Comment:
```jsx

        <PopupWithForm
          isOpen={popupState.isSigninPopupOpen}
          onSubmit={handleSignin}
          isValid={true}
          formName='signin'
          title='Sign in'
          buttonText='Sign in'
          redirectText='Sign up'
          handleRedirect={showSignUp}
          responseError={responseError}
        >
          <AuthForm />
        </PopupWithForm>

```


Please, create `Login` and `Register` and put the whole markup inside them.  You need to have those components in your app


so, here will be only 


```jsx

      {popupState.isSigninPopupOpen && (
        <Login
          isOpen={popupState.isSigninPopupOpen}
          onSubmit={handleSignin}
          handleRedirect={showSignUp}
          responseError={responseError}
        />
      )}

```